android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/details/AncDetailsViewModel.kt

```diff
@@ -46,15 +50,74 @@ class AncDetailsViewModel(
     return patientDemographics
   }
 
-  fun fetchCarePlan(cJson: String?): LiveData<List<CarePlanItem>> {
+  fun fetchCarePlan(): LiveData<List<CarePlanItem>> {
     val patientCarePlan = MutableLiveData<List<CarePlanItem>>()
     viewModelScope.launch(dispatcher.io()) {
-      val listCarePlan = ancPatientRepository.fetchCarePlan(patientId = patientId, cJson)
-      patientCarePlan.postValue(listCarePlan)
+      val listCarePlan = ancPatientRepository.searchCarePlan(id = patientId)
+      val listCarePlanItem = ancPatientRepository.fetchCarePlanItem(listCarePlan, patientId)
+      patientCarePlan.postValue(listCarePlanItem)
     }
     return patientCarePlan
   }
 
+  fun fetchObservation(): LiveData<AncOverviewItem> {
+    val patientAncOverviewItem = MutableLiveData<AncOverviewItem>()
+    val ancOverviewItem = AncOverviewItem()
+    viewModelScope.launch(dispatcher.io()) {
+      val listObservation = ancPatientRepository.fetchObservations(patientId = patientId)
+      if (listObservation.isNotEmpty()) {
+        for (i in listObservation.indices) {
+          if (listObservation[i].code != null)
+            if (listObservation[i].code.coding != null)
+              if (listObservation[i].code.coding.isNotEmpty())
+                for (j in listObservation[i].code.coding.indices) {
+                  val coding = listObservation[i].code.coding[j] as Coding
+                  if (coding.display != null) {
+                    if (coding.display.isNotEmpty()) {
+                      when {
+                        coding.display.lowercase().contains("edd") ->
```

Comment:
can you query the ancPatientRepository for this directly? instead of adding the looping, this will get too expensive with a large number of observations

Reply:
nb if we query directly the ifs above won't matter

Reply:
currently, pregnancy related information is being stored as observations actually. @maimoonak can clarify this information.

Reply:
can we directly query observations?

Reply:
Currently I am fetching observations of a single patient and searching for EDD, GA and other information.

Reply:
Yes I understand, can we query for Observations directly? That would eliminate needing to search through the returned Observations

Reply:
```
fhirEngine.search<Observation>{
     val coding =  Coding().apply {
       this.code = "my-obs-code"
       this.system = "http://my-obs-code-system"
      }
      filter(Observation.CODE, coding)

      // for patient filter use extension created
      filterByPatient(Observation.SUBJECT, "your-patient-id")
    }
```

To make it more dynamic you can check how ANC and Family is filtered here  
https://github.com/opensrp/fhircore/blob/main/android/anc/src/main/assets/register_configurations.json
and used here https://github.com/opensrp/fhircore/blob/main/android/anc/src/main/java/org/smartregister/fhircore/anc/data/anc/AncPatientRepository.kt#L78

Reply:
Plz pass codeable-concept instead of Coding if Observation code type is codeable concept. That is almost similar and you can use coding..asCodeableConcept() as well